rust/agama-server/src/questions/web.rs

```diff
@@ -46,6 +49,46 @@ impl<'a> QuestionsClient<'a> {
         })
     }
 
+    pub async fn create_question(&self, question: Question) -> Result<Question, ServiceError> {
+        // TODO: ugly API is caused by dbus method to create question. It can be changed in future as DBus is internal only API
+        let generic = &question.generic;
+        let options: Vec<&str> = generic.options.iter().map(String::as_ref).collect();
+        let data: HashMap<&str, &str> = generic
+            .data
+            .iter()
+            .map(|(k, v)| (k.as_str(), v.as_str()))
+            .collect();
+        let path = if question.with_password.is_some() {
+            self.questions_proxy
+                .new_with_password(
+                    &generic.class,
+                    &generic.text,
+                    &options,
+                    &generic.default_option,
+                    data,
+                )
+                .await?
+        } else {
+            self.questions_proxy
+                .new_question(
+                    &generic.class,
+                    &generic.text,
+                    &options,
+                    &generic.default_option,
+                    data,
+                )
+                .await?
+        };
+        let mut res = question.clone();
+        // we are sure that regexp is correct, so use unwrap
+        let id_matcher = Regex::new(r"/(?<id>\d+)$").unwrap();
+        let Some(id_cap) = id_matcher.captures(path.as_str()) else {
+            panic!("Id not in path")
+        }; // TODO: better error if path does not contain id
```

Comment:
Does this code crash the server?

Reply:
yeap, it terminates...but it should never happen as new question if method succeed should always return id as part of path